app/src/main/java/com/teamounce/ounce/main/BottomSheetAdapter.kt

```diff
@@ -1,52 +1,101 @@
 
-class BottomSheetAdapter : RecyclerView.Adapter<BottomSheetAdapter.BottomSheetViewHolder>() {
-
-    var settingCareCatData = mutableListOf<SettingCareCatData>()
-    var lastSelectionPosition = -1
-
-    override fun getItemCount(): Int {
-        return settingCareCatData.size
-    }
+class BottomSheetAdapter(private var context: Context) :
+    RecyclerView.Adapter<BottomSheetAdapter.BottomSheetViewHolder>() {
+    val mContext = getActivity(context)
+    var bottomSheetProfileData = mutableListOf<BottomSheetProfileData>()
+    private var checkedRadioButton: CompoundButton? = null
+    private var mSelectedItem = -1
+    val bottomSheetFragment = BottomSheetFragment()
 
     override fun onCreateViewHolder(
         parent: ViewGroup,
         viewType: Int
     ): BottomSheetViewHolder {
-        val binding = ItemBottomMainBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        val binding =
+            ItemBottomMainBinding.inflate(LayoutInflater.from(parent.context), parent, false)
         return BottomSheetViewHolder(binding)
     }
 
     override fun onBindViewHolder(holder: BottomSheetViewHolder, position: Int) {
-        holder.bind(settingCareCatData[position])
+        holder.bind(bottomSheetProfileData[position], position, mSelectedItem)
+        holder.binding.catSelectBtn.setOnCheckedChangeListener(checkedChangedListener)
 
-            holder.binding.catSelectBtn.isChecked = lastSelectionPosition == position
+        if (holder.binding.catSelectBtn.isChecked) {
+            checkedRadioButton = holder.binding.catSelectBtn
+        }
+    }
 
+    override fun getItemCount(): Int {
+        return bottomSheetProfileData.size
+    }
 
+    fun changeCat(position: Int) {
+        if (mContext is MainActivity) {
+            mContext.setMainViewRetrofit(bottomSheetProfileData[position].catIndex)
+            mContext.bottomSheetFragment.dismiss()
         }
+    }
 
-    inner class BottomSheetViewHolder(var binding: ItemBottomMainBinding) : RecyclerView.ViewHolder(binding.root){
+    private val checkedChangedListener =
+        CompoundButton.OnCheckedChangeListener { compoundButton, isChecked ->
+            checkedRadioButton?.apply {
+                setChecked(!isChecked)
+            }
+            checkedRadioButton = compoundButton.apply {
+                setChecked(isChecked)
+                bottomSheetFragment.fragmentManager?.executePendingTransactions()
+                bottomSheetFragment.dialog?.setOnDismissListener {
+                    it.dismiss()
+                }
+            }
 
-        fun bind(settingCareCatData: SettingCareCatData) {
-            binding.settingCareCatData = settingCareCatData
+        }
 
-            binding.catSelectBtn.setOnClickListener {
-                lastSelectionPosition = adapterPosition
-                notifyDataSetChanged()
+    inner class BottomSheetViewHolder(var binding: ItemBottomMainBinding) :
```

Comment:
inner 클래스와 일반적인 중첩 클래스는 코틀린에서 그 의미가 달라. 어떤 부분이 다를까? 난 굳이 이 클래스가 inner일 필요가 없을 것 같아
```suggestion
    class BottomSheetViewHolder(var binding: ItemBottomMainBinding) :
```